provider/pkg/gen/nodejs-templates/helm/v3/helm.ts

```diff
@@ -0,0 +1,356 @@
+ *   fetchOpts:{
+ *     repo: "https://kubernetes-charts.storage.googleapis.com/",
+ *   },
+ * });
+ * ```
+ * ### Chart with Transformations
+ *
+ * ```typescript
+ * import * as k8s from "@pulumi/kubernetes";
+ *
+ * const nginxIngress = new k8s.helm.v3.Chart("nginx-ingress", {
+ *   chart: "nginx-ingress",
+ *   version: "1.24.4",
+ *   fetchOpts:{
+ *     repo: "https://kubernetes-charts.storage.googleapis.com/",
+ *   },
+ *   transformations: [
+ *     // Make every service private to the cluster, i.e., turn all services into ClusterIP instead of LoadBalancer.
+ *     (obj: any, opts: pulumi.CustomResourceOptions) => {
+ *       if (obj.kind === "Service" && obj.apiVersion === "v1") {
+ *         if (obj.spec && obj.spec.type && obj.spec.type === "LoadBalancer") {
+ *           obj.spec.type = "ClusterIP";
+ *         }
+ *       }
+ *     },
+ *
+ *     // Set a resource alias for a previous name.
+ *     (obj: any, opts: pulumi.CustomResourceOptions) => {
+ *     if (obj.kind === "Deployment") {
+ *       opts.aliases = [{ name: "oldName" }]
+ *     },
+ *
+ *     // Omit a resource from the Chart by transforming the specified resource definition to an empty List.
+ *     (obj: any, opts: pulumi.CustomResourceOptions) => {
+ *     if (obj.kind === "Pod" && obj.metadata.name === "test") {
+ *       obj.apiVersion = "v1"
+ *       obj.kind = "List"
+ *     },
+ *   ],
+ * });
+ * ```
+ */
+export class Chart extends yaml.CollectionComponentResource {
+    /**
+     * Create an instance of the specified Helm chart.
+     * @param releaseName Name of the Chart (e.g., nginx-ingress).
+     * @param config Configuration options for the Chart.
+     * @param opts A bag of options that control this resource's behavior.
+     */
+    constructor(
+        releaseName: string,
+        config: ChartOpts | LocalChartOpts,
+        opts?: pulumi.ComponentResourceOptions
+    ) {
+        if (config.resourcePrefix !== undefined) {
+            releaseName = `${config.resourcePrefix}-${releaseName}`
+        }
+        const aliasOpts: pulumi.ComponentResourceOptions = {...opts, aliases: [{type:"kubernetes:helm.sh/v2:Chart"}]}
+        super("kubernetes:helm.sh/v3:Chart", releaseName, config, aliasOpts);
+
+        const allConfig = pulumi.output(config);
+
+        (<any>allConfig).isKnown.then((isKnown: boolean) => {
+            if (!isKnown) {
+                // Note that this can only happen during a preview.
+                pulumi.log.info("[Can't preview] all chart values must be known ahead of time to generate an " +
+                    "accurate preview.", this);
+            }
+        });
+
+        this.resources = allConfig.apply(cfg => {
+            return this.parseChart(cfg, releaseName, opts)
+        });
+    }
+
+    parseChart(config: ChartOpts | LocalChartOpts, releaseName: string, opts?: pulumi.ComponentResourceOptions) {
+        const blob = {
+            ...config,
+            releaseName,
+
+            toJSON() {
```

Comment:
Not sure if there's a simpler way to handle the translation from the input interfaces to the JSON string, but I had to add cases for all of the camelCase fields.

Reply:
The invoke call is expecting a JSON string that it unmarshals to the Helm structs in the provider, so I need the keys to match. I'm not sure if there's a better way to handle this, but the data was complex enough that I thought JSON was nicer than using individual fields.

Reply:
> Also, TIL about toJSON(). The embedding of logic on the config props like this is stylistically new to me.

Courtesy of https://www.executeprogram.com/courses/modern-javascript/lessons/customizing-json-serialization